go-controller/pkg/cluster/master.go

```diff
@@ -0,0 +1,143 @@
+package cluster
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+
+	"github.com/golang/glog"
+
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+	kapi "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/origin/pkg/util/netutils"
+)
+
+func (cluster *OvnClusterController) StartClusterMaster(masterNodeName string) error {
+	clusterNetwork := cluster.ClusterIPNet
+	hostSubnetLength := cluster.HostSubnetLength
+
+	subrange := make([]string, 0)
+	existingNodes, err := cluster.Kube.GetNodes()
+	if err != nil {
+		glog.Errorf("Error in initializing/fetching subnets: %v", err)
+		return err
+	}
+	for _, node := range existingNodes.Items {
+		hostsubnet, ok := node.Annotations[OVN_HOST_SUBNET]
+		if ok {
+			subrange = append(subrange, hostsubnet)
+		}
+	}
+	masterSwitchNetwork, err := calculateMasterSwitchNetwork(clusterNetwork.String(), hostSubnetLength)
```

Comment:
Right. Subnet allocation happens only in this code anyway which is only called in the master mode.
It's not intended that we run two copies of this code in parallel, but we could keep locks in place to check for that. Right now that feels more like paranoia than perfection.